db/migrate/20170517222310_change_annotation_author_to_not_null.rb

```diff
@@ -9,6 +9,10 @@ def up
       'SELECT id FROM users ORDER BY created_at ASC LIMIT 1'
     ).first['id']
 
+    if default_author.nil?
```

Comment:
This line itself raises an error—there’s no `default_author` variable. To do this, you still need to change where `default_author_id` is defined above, so you have:

```rb
default_author = do_sql('same query').first
```

…and where it is used below:

```rb
do_sql('some query', default_author['id'])
```

This is also the reason you were getting the `undefined method '[]' for nil:NilClass` error—`do_sql(...).first` was `nil` (no users in the DB), so you can’t call `['id']` on it. To work your way around this, you can:

```sh
rake db:create
rake db:migrate VERSION=20170508221326 # stop before the author not null migration
rails c
# In the Rails console, create a new user like so, then quit:
# User.create(email: 'someone@example.com', password: 'password', confirmed_at: Time.now)
rake db:migrate
```

(You could also change this migration to *create* a user if one is not present.)

Reply:
Cool.  Sorry for so much back-and-forth on this; context switching between my work and this means I'm missing some easy stuff.  I'll fix this bug and run again.  Also, thanks for the explanation on db:setup; I'm used to working in a local env where we have a make task that re-migrates every time, I forgot that wasn't default behavior. :)  Thanks for your patience and your help, will fix and push again tonight hopefully!